src/elements/content-sidebar/ActivitySidebar.js

```diff
@@ -405,14 +441,23 @@ class ActivitySidebar extends React.PureComponent<Props, State> {
      *
      * @param {boolean} shouldDestroy true if the api factory should be destroyed
      */
-    fetchFeedItems(shouldRefreshCache: boolean = false, shouldDestroy: boolean = false) {
+    fetchFeedItems(
+        shouldRefreshCache: boolean = false,
+        shouldDestroy: boolean = false,
+        successCallback: (feedItems: FeedItems) => void = noop,
+        errorCallback: (feedItems: FeedItems) => void = noop,
+    ) {
         const { file, api, features } = this.props;
         const shouldShowAppActivity = isFeatureEnabled(features, 'activityFeed.appActivity.enabled');
+        const successWrapper = feedItems =>
+            [successCallback, this.fetchFeedItemsSuccessCallback].forEach(callback => callback(feedItems));
```

Comment:
This seems a little unfortunate. Should we introduce promises or something rather than relying on callbacks for everything?